hashmail_server.go

```diff
@@ -284,12 +306,14 @@ func newStream(id streamID, limiter *rate.Limiter,
 // ReturnReadStream returns the target read stream back to its holding channel.
 func (s *stream) ReturnReadStream(r *readStream) {
 	s.readStreamChan <- r
+	s.status.streamReturned()
```

Comment:
Can we add a simple unit test to make sure these are counted at the correct moment? I'm not sure when those channels actually are sent to. Just to avoid surprises.

Reply:
good idea - I added a test, let me know if this is what you had in mind :)